fsmonitor-settings.c

```diff
@@ -58,17 +58,16 @@ static int check_deprecated_builtin_config(struct repository *r)
 	 * set to true, set the appropriate mode and return 1 indicating that
 	 * the check resulted the config being set by this (deprecated) setting.
 	 */
-	if(!repo_config_get_bool(r, "core.useBuiltinFSMonitor", &core_use_builtin_fsmonitor)) {
+	if(!repo_config_get_bool(r, "core.useBuiltinFSMonitor", &core_use_builtin_fsmonitor) &&
+	   core_use_builtin_fsmonitor) {
```

Comment:
some compilers will complain that core_use... might be uninitialized (if it can't do full cross-functional flow analysis),  so maybe an =0 at the top.

Reply:
That's a good point. The second half of the `&&` is only evaluated if `repo_config_get_bool()` has assigned the value, but there is no ability to see that within this file.